lib/SimpleHttpClient/ConnectionCache.cpp

```diff
@@ -50,21 +51,22 @@ ConnectionLease::~ConnectionLease() {
 ConnectionLease::ConnectionLease(ConnectionLease&& other) noexcept
     : _cache(other._cache),
       _connection(std::move(other._connection)),
-      _preventRecycling(other._preventRecycling) {}
+      _preventRecycling(
+          other._preventRecycling.load(std::memory_order_relaxed)) {}
 
 ConnectionLease& ConnectionLease::operator=(ConnectionLease&& other) noexcept {
   if (this != &other) {
     _cache = other._cache;
     _connection = std::move(other._connection);
-    _preventRecycling = other._preventRecycling;
+    _preventRecycling = other._preventRecycling.load(std::memory_order_relaxed);
   }
   return *this;
 }
 
 void ConnectionLease::preventRecycling() noexcept {
   // this will prevent the connection from being inserted back into the
   // connection cache
-  _preventRecycling = true;
+  _preventRecycling.store(true);  // data race write and read
```

Comment:
Is it on purpose that this store is `seq_cst`, while both loads above are `relaxed`?

Reply:
There is an implicit load happening in the `ConnectionLease` destructor (line 46), `!_preventRecycling`, so there we use `seq_cst` for loading. At least in devel we do it in the same way, but now that you mention it, I don't see why we should not use at least `aquire` - `release` semantics here instead of `seq_cst`.